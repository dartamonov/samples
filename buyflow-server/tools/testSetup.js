// Tests are placed alongside files under test.
// This file does the following:
// 1. Sets the environment to 'production' so that
//    dev-specific babel config in .babelrc doesn't run.
// 2. Disables Webpack-specific features that Mocha doesn't understand.
// 3. Registers babel for transpiling our code for testing.

// This assures the .babelrc dev config (which includes
// hot module reloading code) doesn't apply for tests.
// process.env.NODE_ENV = 'production';
// process.env.NODE_ENV = 'test';

// Register babel so that it will transpile ES6 to ES5
// before our tests run.
require('babel-register')();
//require("babel-polyfill");
require("babel-jest");
