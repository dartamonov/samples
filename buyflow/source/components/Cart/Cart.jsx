import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';
import _ from 'lodash';
import Price from '../Price/Price.jsx';

class Cart extends Component {
  render() {
    if (_.isEmpty(this.props.cart)) {
      return null;
    }

    const {
      continueAction,
      cart
    } = this.props;
    const valid = cart.monthlyTotal > 0;
    const nextBtnClass = {
      'x-button--solid' : valid !== false,
      'x-button--disabled' : valid === false
    };

    return (
      <div className="hg-cart">
        <div className="hg-cart__panel">
          <div className="hg-cart__panel-content">
            <h2 className="hg-cart__panel-title x-heading3">Monthly Total</h2>
          </div>
          <div className="hg-cart__panel-content">
            <Price price={cart.monthlyTotal} />
          </div>
        </div>
        <div className="hg-cart__panel">
          <button type="button" className={classNames(nextBtnClass)} onClick={continueAction}>Review</button>
        </div>
      </div>
    );
  }

}

Cart.propTypes = {
  continueAction: PropTypes.func,
  cart: PropTypes.object.isRequired
};

export default Cart;
