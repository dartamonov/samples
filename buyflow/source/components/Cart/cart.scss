@import "./cartpanel.scss";

// Hide primary cart on show summary on mobile
// Do the reverse above mobile
.hg-cart {
  left: 0;
  right: 0;
  top: calc(100% - 75px); //minus the height of visible header
  bottom: 0;
  z-index: 300;  //TODO: DeEscalate z-index radios have 200 currently
  background-color: #F2F4F4;
  transition: top 0.5s ease-in-out;
  will-change: top;

  @include breakpoint($bpLarge) {
    will-change: auto;
    position: static;
    background: none;
    overflow: visible;
  }

  &.hg-cart--summary {
    top: 0; //minus the height of visible header
  }
}

.hg-cart__content {
  overflow: hidden;
}

.hg-cart__content--primary {
  display: none;

  @include breakpoint($bpLarge) {
    display: block;
  }
}

.hg-cart__content--summary {
  display: none;

  .hg-cart--summary & {
    display: block;

    @include breakpoint($bpLarge) {
      display: none;
    }
  }
}

.hg-cart__summary-content {
  display: none;
  overflow: auto;
  max-height: 80vh; // we want it large enough to show the maximum amount of content, but not so large that it pushes the "Need help?" message off the screen.

  @include breakpoint($bpLarge) {
    display: block;
  }

  .hg-cart--summary & {
    display: block;
  }
}

.hg-cart__summary-header {
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 10;

  @include breakpoint($bpLarge) {
    display: none;
  }

  .hg-cart--summary & {
    display: none;
  }
}

.hg-cart__panel--nav {
  display: none;
  text-align: center;
  background: $white;
  margin-bottom: unit-calc(3px);

  @include breakpoint($bpLarge) {
    display: flex;
  }

  .hg-cart--review & {
    display: flex;
  }
}

.hg-cart__panel--mobile-header {
  display: none;

  .hg-cart--summary & {
    display: block;

    @include breakpoint($bpLarge) {
      display: none;
    }
  }
}

.hg-cart__toggle-summary {
  @include breakpoint($bpLarge) {
    display: none;
  }

  .hg-cart--has-summary-on-desktop & {
    @include breakpoint($bpLarge) {
      display: inline-block;
    }
  }
}

.hg-cart__toggle-summary--icon {
  @include breakpoint($bpLarge) {
    transition: transform 0.5s ease-in-out;
    transform: rotate(0);
  }

  .hg-cart--summary & {
    @include breakpoint($bpLarge) {
      transform: rotate(45deg);
    }
  }
}

.hg-cart__panel--mobile-footer {
  @include breakpoint($bpLarge) {
    display: none;
  }
}

.hg-cart__content--monthly-total {
  display: none;

  @include breakpoint($bpLarge) {
    display: flex;
  }
}

.hg-cart__content--monthly-total-small {
  @include breakpoint($bpLarge) {
    display: none;
  }
}

.hg-cart__items {
  @include media-breakpoint-up('md') {
    max-height: 156px;
    overflow: auto;
  }
}

.hg-cart__footer {
}

.hg-cart__footer--center {
  justify-content: center;
  min-height:20px;
}
