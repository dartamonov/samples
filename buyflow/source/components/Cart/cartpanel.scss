.hg-cart__panel {
  position: relative;
  display: flex;
  background: $white;
  justify-content: space-between;
  align-items: center;
  padding: unit-calc(8px) unit-calc(20px);
  margin-bottom: unit-calc(3px);
  min-height: 60px;
}

.hg-cart__panel--overridden,
.hg-cart__panel--overridden .x-body3,
.hg-cart__panel--one-time .hg-cart__panel-desc {
  font-style: italic;
  color: $grey36;
}

.hg-cart__panel--removable,
.hg-cart__panel--due-today {
  border-left: 5px solid $blueSky;
  padding-left: unit-calc(15px);
}

.hg-cart__panel--removable,
.hg-cart__panel--included {
  background: $coolGrey1;
}

.hg-cart__offer-price {
  .hg-price {
    text-align: right;
  }
}

.hg-cart__panel--one-time {
  border-left: 5px solid $yellowPumpkin;

  .hg-cart__panel-title {
    margin-bottom: 0;
  }
}

.hg-cart__panel--due-today {
  margin-bottom: 0;
}

.hg-cart__panel-title {
  margin-bottom: unit-calc(5px);
}

.hg-cart__panel-desc {
  margin-bottom: unit-calc(3px);
}

.hg-cart__panel--mobile-footer {
  margin-bottom: 0;
  background: linear-gradient(to bottom, rgba(255,255,255,0.5) 0%,rgba(255,255,255,0.75) 18%,rgba(255,255,255,1) 41%,rgba(255,255,255,1) 100%);
  .x-button--solid {
    padding: unit-calc(5px);
    min-width: unit-calc(65px);
  }
}

.hg-cart__panel--mobile-header {
  border-bottom: 3px solid $grey9;
}

.hg-cart__panel-link {
  @extend a;
}

.hg-cart__offer-price {
  margin-bottom: 0;
}

.hg-cart__panel-content--price {
  display: flex;
  align-items: center;
}

.hg-cart__panel-content-link-layout {
  &,
  &:hover,
  &:focus {
    display: flex;
    width: 100%;
  }

  .hg-cart__panel-content--centered {
    flex: 1;
    text-align: center;
    margin-right: unit-calc(-17px);
  }

  .hg-cart__panel-content--side {
    margin-left: auto;
  }
}

.hg-cart__panel-content-icon {
  display: none;
  position: absolute;
  top: 26px;
  right: -30px;

  .hg-cart--has-summary-on-desktop & {
    @include breakpoint($bpMedium) {
      display: block;
    }
  }

  button {
    background: $white;
    border: 1px solid $blueSky;
    padding: 5px;
    border-radius: 50%;
  }

  svg {
    cursor: pointer;
    pointer-events: none;
    display: block;
  }
}

.hg-cart__panel-icon {
  @include font-size(10px);
  color: $white;
  border: 0;
  padding: unit-calc(0px 6px 2px);
  border-radius: unit-calc(10px);
  background: $blueSky;
  margin-left: unit-calc(12px);
}

.hg-cart__panel-content {
  .hg-cart__panel-title,
  .hg-cart__panel-desc,
  .hg-cart__offer-price {
    &:last-child {
      margin-bottom: 0;
    }
  }
}

.hg-cart__panel--nav-previous {
  display: none;
  @include breakpoint($bpXlarge) {
    display: block;
  }
}
