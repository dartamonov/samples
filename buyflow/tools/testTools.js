import renderer from 'react-test-renderer';

export const takeSnapshot = x => renderer.create(x).toJSON();