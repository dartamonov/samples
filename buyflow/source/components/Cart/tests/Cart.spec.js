import React from 'react';
import Cart from '../Cart.jsx';
import {takeSnapshot} from '../../../tools/testTools';

describe('Cart component', () => {
  it('handles normal price', () => {
    const cartData = { monthlyTotal: 10.95 };
    const snapshot = takeSnapshot(
      <Cart cart={cartData} />
    );

    expect(snapshot).toMatchSnapshot();
  });
});
