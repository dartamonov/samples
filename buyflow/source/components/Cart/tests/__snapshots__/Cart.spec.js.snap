// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component handles normal price 1`] = `
<div
  className="hg-cart"
>
  <div
    className="hg-cart__panel"
  >
    <div
      className="hg-cart__panel-content"
    >
      <h2
        className="hg-cart__panel-title x-heading3"
      >
        Monthly Total
      </h2>
    </div>
    <div
      className="hg-cart__panel-content"
    >
      <span
        className="hg-price"
      >
        <span
          className="hg-price__icon"
        >
          $
        </span>
        <span
          className="hg-price__dollars"
        >
          10
        </span>
        <span
          className="hg-price__cents"
        >
          .95
        </span>
        <span>
          /mo
        </span>
      </span>
    </div>
  </div>
  <div
    className="hg-cart__panel"
  >
    <button
      className="x-button--solid"
      type="button"
    >
      Review
    </button>
  </div>
</div>
`;
